### PR TITLE
Use official typing package for `trio`

### PR DESCRIPTION
**Pull request by michaeloliverx**

### CODE REVIEW

**Comment by michaeloliverx** [comment]
Refs https://github.com/encode/httpcore/pull/618

File: httpcore/backends/trio.py [hunk 1]
```diff
@@ -118,14 +120,15 @@ async def connect_tcp(
...
         with map_exceptions(exc_map):
             with trio.fail_after(timeout_or_inf):
                 stream: trio.abc.Stream = await trio.open_tcp_stream(
-                    host=host, port=port, **kwargs
+                    host=host,
+                    port=port,
+                    # Trio supports 'local_address' from 0.16.1 onwards.
+                    # We only include the keyword argument if a local_address
+                    # argument has been passed.
+                    local_address=local_address,
```
Comment (michaeloliverx) [inline httpcore/backends/trio.py:131]: Had to inline this as mypy could only infer kwargs as `dict[str, str]`, this is the same behaviour as before though pretty sure.

Comment (tomchristie) [reply httpcore/backends/trio.py:131]: Gotcha.

It is a minor behavioural change in that it means our trio dependancy is now *strictly* >= 0.16.1, whereas before our dependancy only required >= 0.16.1 *if* `local_address` was being used.

But I think we're okay with that... https://pypi.org/project/trio/#history

**Review by tomchristie — APPROVED** [review]

File: httpcore/backends/trio.py (hunk 1 above)
Comment (michaeloliverx) [reply httpcore/backends/trio.py:131]: Ahh I get it now, we could split the calls if required but probably not needed as 0.16.1 is quite old now.

If we are going forward with the new behaviour shall I fix the comment?

Comment (tomchristie) [reply httpcore/backends/trio.py:131]: I think we can just drop the comment yup.

**Pull request merged**